### PR TITLE
Operation/initial obc implementation

### LINKED ISSUE #143: On-Board-Computer
All functions or implementations related to the obc should be talked about here

This module should do the following 

- [x] Implement basic system operations
- [x] Collect telemetry from the board
- [ ] Pet the watchdog on board

### PR DESCRIPTION
**Pull request by ChrisAmaya**
A fairly simple addition to the code

### CODE REVIEW

**Review by matthew-buglass — DISMISSED** [review]

**Review by TheDopplerEffects — DISMISSED** [review]

File: radsat-sk/operation/subsystems/RObc.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
+ * @file RObc.h
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RObc.c]: Incorrect file type

File: radsat-sk/operation/subsystems/RObc.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * @return	Error code; 0 for success, otherwise see hal/errors.h.
+ */
+int obcTelemetry(obcTelemetry_t* obcTelemetryBuffer){
+	float temperature = 0;
+	int error = RTC_getTemperature(&temperature);
+
+	if (error != SUCCESS) {
+		return error;
+	}
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RObc.c]: Need to report error to the Error Manager (once it's available)

File: radsat-sk/operation/subsystems/RObc.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Retrieve Temperature telemetry from OBC
+ *
+ * @return	Error code; 0 for success, otherwise see hal/errors.h.
+ */
+int obcTelemetry(obcTelemetry_t* obcTelemetryBuffer){
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RObc.c]: The word "buffer" doesn't make a ton of sense in this context; I would just call it "telemetry"

File: radsat-sk/operation/subsystems/RObc.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Pet the watchdog on the OBC
+ *
+ * @return	N/A
+ */
+void obcPetWatchdogs(void){
+	WDT_forceKick();
+}
+
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RObc.c]: This function probably isn't necessary; we already have `WDT_startWatchdogKickTask` provided by the HAL which is supposed to start a FreeRTOS task to do all of the kicking for us. At the very least, this function should mention the existence of that HAL function.

File: radsat-sk/operation/subsystems/RObc.h [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#include <stdlib.h>
+
+
+/***************************************************************************************************
+                                            DEFINITIONS
+***************************************************************************************************/
+
+typedef struct _obcTelemetry_t {
+	float temperature;
+} obcTelemetry_t;
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RObc.h]: Incorrect style name for the struct. Proper name would be `obc_telemetry_t`

File: radsat-sk/operation/subsystems/RObc.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+int obcTelemetry(obcTelemetry_t* obcTelemetryBuffer);
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RObc.h]: See other commend on name of argument; would just call it `telemetry` instead

**Review by tyrelkostyk — DISMISSED** [review]

File: radsat-sk/operation/subsystems/RObc.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Retrieve Temperature telemetry from OBC
+ *
+ * @return	Error code; 0 for success, otherwise see hal/errors.h.
+ */
+int obcTelemetry(obcTelemetry_t* telemetry){
```
Comment (ATGr8) [inline radsat-sk/operation/subsystems/RObc.c]: add space between ) and {

File: radsat-sk/operation/subsystems/RObc.h [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+ * @file RObc.h
+ * @date April 2, 2022
+ * @author Matthew Buglass (mab839)
+ */
+
+#ifndef ROBC_H_
+#define ROBC_H_
+
+#include <stdlib.h>
+
```
Comment (ATGr8) [inline radsat-sk/operation/subsystems/RObc.h:11]: remove whitespace

File: radsat-sk/operation/subsystems/RObc.h [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+int obcTelemetry(obcTelemetry_t* telemetry);
+void obcPetWatchdogs(void);
+
+
+
```
Comment (ATGr8) [inline radsat-sk/operation/subsystems/RObc.h]: remove whitespace

File: radsat-sk/operation/subsystems/RObc.h [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+/***************************************************************************************************
+                                            DEFINITIONS
+***************************************************************************************************/
+
+typedef struct obc_telemetry_t {
+	float temperature;
+} obcTelemetry_t;
+
+
+
```
Comment (ATGr8) [inline radsat-sk/operation/subsystems/RObc.h:21]: remove whitespace

File: radsat-sk/operation/subsystems/RObc.h [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+ * @date April 2, 2022
+ * @author Matthew Buglass (mab839)
+ */
+
+#ifndef ROBC_H_
+#define ROBC_H_
+
+#include <stdlib.h>
+
+
```
Comment (ATGr8) [inline radsat-sk/operation/subsystems/RObc.h:12]: remove whitespace

**Comment by ATGr8** [comment]
petWatchDog function is defined in .h file but not implemented in .c file

File: radsat-sk/operation/subsystems/RObc.c [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+#include <hal/Timing/RTC.h>
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Retrieve Temperature telemetry from OBC
+ *
+ * @return	Error code; 0 for success, otherwise see hal/errors.h.
```
Comment (ATGr8) [inline radsat-sk/operation/subsystems/RObc.c:20]: remove whitespace between @return and "Error code"

**Review by ATGr8 — APPROVED** [review]
Looks good!

**Pull request merged**